### PR TITLE
Shavar dockerflow

### PR DESCRIPTION
**Pull request by ckolos**
@groovecoder r?

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+machine:
+  services:
+    - docker
+
+dependencies:
+  cache_directories:
+    - "~/docker"
+  override:
+    - docker info
+    - I="image-$(date +%j).tar";if [[ -e ~/docker/$I ]]; then echo "Loading $I";docker load -i ~/docker/$I;fi
```
Comment (groovecoder) [inline circle.yml]: So this attempts to load an existing `image-<day-of-year>.tar` if one already exists ...

File: circle.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        "$CIRCLE_SHA1"
+        "$CIRCLE_TAG"
+        "$CIRCLE_PROJECT_USERNAME"
+        "$CIRCLE_PROJECT_REPONAME"
+        "$CIRCLE_BUILD_URL"
+        > version.json
+    - cp version.json $CIRCLE_ARTIFACTS
+    - docker build -t app:build .
+    - >
+      I="image-$(date +%j).tar"; mkdir -p ~/docker; rm ~/docker/*; docker save app:build > ~/docker/$I; ls -l ~/docker
```
Comment (groovecoder) [inline circle.yml]: ... and then this over-writes the `images-<day-of-year>.tar` file after the new build?

**Comment by groovecoder** [comment]
This is looking good to me. I was able to build the docker image locally, run it, and curl the list endpoint:
```
(shavar)MacBook-Pro:shavar lcrouch$ curl -d " " --trace-ascii - http://192.168.99.100:2376/list?client=foo\&appver=1\&pver=2.2
== Info:   Trying 192.168.99.100...
== Info: Connected to 192.168.99.100 (192.168.99.100) port 2376 (#0)
=> Send header, 185 bytes (0xb9)
0000: POST /list?client=foo&appver=1&pver=2.2 HTTP/1.1
0032: Host: 192.168.99.100:2376
004d: User-Agent: curl/7.43.0
0066: Accept: */*
0073: Content-Length: 1
0086: Content-Type: application/x-www-form-urlencoded
00b7:
=> Send data, 1 bytes (0x1)
0000:
== Info: upload completely sent off: 1 out of 1 bytes
<= Recv data, 7 bytes (0x7)
0000: .......
== Info: Connection #0 to host 192.168.99.100 left intact
```
Should I do end-to-end test to create the lists and make they are all served correctly too?

**Comment by groovecoder** [comment]
Note: `http://192.168.99.100:2376` is my docker machine vm `DOCKER_HOST`

File: circle.yml (hunk 2 above)
Comment (ckolos) [inline circle.yml]: The first iteration loads it if it already exists; this speeds up the process for multiple builds. The second iteration creates that "cache"

**Comment by ckolos** [comment]
I would say no to that only b/c the lists aren't meant to be accessible from the circle build servers. The fact that the api is working is all that's required here for the image to be "valid"

**Comment by groovecoder** [comment]
Then this looks good to me. Anything to add before I merge?

**Comment by ckolos** [comment]
Nope, good for now!

**Pull request merged**